magefiles/publish.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+)
+
+type BumpVersion mg.Namespace
+
+// Ts bumps the version of the typescript packages with the given version.
+func (BumpVersion) Ts(version string) error {
+	fmt.Println("🔖 Bump typescript packages version")
+
+	ensureYarn()
+
+	err := isVersionTagValid(version)
+	if err != nil {
+		return err
+	}
+
+	packages, err := os.ReadDir(TS_DIR)
+	if err != nil {
+		return fmt.Errorf("failed to read typescript directory '%s': %w", TS_DIR, err)
+	}
+
+	for _, pkg := range packages {
+		if !pkg.IsDir() {
+			continue
+		}
+		fmt.Printf("    ➡️ Bumping %s to version %s\n", pkg.Name(), version)
+		err := sh.Run("yarn",
+			"--cwd", filepath.Join(TS_DIR, pkg.Name()),
+			"version",
+			"--new-version", version,
+			"--allow-same-version",
+			"--no-git-tag-version")
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Go bumps the version of the go packages with the given version.
+func (BumpVersion) Go(version string) error {
+	fmt.Println("🔖 Bump go packages version")
+
+	err := isVersionTagValid(version)
```

Comment:
Minor remark: [parse don't validate](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/).

Create a `Version` type and write a function to parse the version from a string. This way, you don’t need to split the string every time you use it.

Reply:
@ccamel I understand and appreciate the principle of 'Parse, don't validate'. However, in the context of this package publishing scenario, implementing a full version parsing seemed to introduce a level of complexity that a simple regex check could handle. That being said, I see the value in this approach and I will proceed with implementing the version parsing as suggested. Thanks 😌.